spring-batch-core/src/main/java/org/springframework/batch/core/jsr/configuration/xml/JsrBeanDefinitionDocumentReader.java

```diff
@@ -182,7 +182,7 @@ private String resolvePropertyValue(String propertyValue) {
 	}
 
 	private String resolveValue(String value) {
-		StringBuffer valueBuffer = new StringBuffer();
+		StringBuilder valueBuffer = new StringBuilder();
```

Comment:
This change creates a compilation error in `jobParameterMatcher.appendReplacement` and `jobParameterMatcher.appendTail`. Those methods accept a `StringBuilder` starting from Java 9 onward. I will revert that change on merge. Please make sure to use Java 8 when building your PR.